mlir/lib/Dialect/Affine/Utils/LoopUtils.cpp

```diff
@@ -1016,8 +1017,12 @@ LogicalResult mlir::affine::loopUnrollByFactor(
     AffineForOp forOp, uint64_t unrollFactor,
     function_ref<void(unsigned, Operation *, OpBuilder)> annotateFn,
     bool cleanUpUnroll) {
-  assert(unrollFactor > 0 && "unroll factor should be positive");
-
+  if (unrollFactor <= 0) {
+    return emitError(
+        forOp.getLoc(),
+        "'-affine-loop-unroll=unroll-factor' cannot be less than or "
+        "equal to zero; pass aborted!");
+  }
```

Comment:
The assert is fine. The utility shouldn't typically emit a message in this situation. The 0 unroll factor should be checked by the user of this utility, which would be the pass, given the crash issue you are addressing. Please put the check and emit message in the pass.